.github/workflows/nightly_run.yml

```diff
@@ -0,0 +1,47 @@
+name: gccrs-testing-nightly-run
+
+on:
+  schedule:
+    - cron: '0 2 * * *' # Run at 2 AM UTC. Does not really matter
+
+jobs:
+  run-testsuites:
+    runs-on: self-hosted
+    container: philberty/gccrs:latest
+    strategy:
+      matrix:
+        testsuite: [rustc-dejagnu, gccrs-parsing, gccrs-rustc-success, gccrs-rustc-success-no-std, gccrs-rustc-success-no-core, blake3, libcore-1.49]
+    steps:
+      - name: Fetch dependencies
+        run: |
+          # FIXME: Switch to `cargo install ftf` on next release, we need the
+          # `result-fmt` feature
+          cargo install --git https://github.com/cohenarthur/ftf
+
+      - uses: actions/checkout@v2
+      - name: Generate testsuites (nightly)
+        run: |
+          cargo build --release
+          git clone https://github.com/rust-gcc/gccrs --depth=1 local_gccrs
+          git submodule update --init
+          target/release/testsuite-adaptor --gccrs rust1 --rustc rustc \
+              --output-dir output-dir-${{ matrix.testsuite }} \
+              --yaml ${{ matrix.testsuite }}.yaml \
+              --rust-path rust --gccrs-path local_gccrs \
+              --passes ${{ matrix.testsuite }}
+
+      - name: Run testsuite
+        run: |
+          echo "{ \"name\": \"${{matrix.testsuite}}\", \"commit\": \"$(cat /GCCRS_BUILD)\", \"date\": \"$(date -I)\", \"results\": " >> ${{ matrix.testsuite }}.json
```

Comment:
Writing this to the file here is a bit ugly, I wouldn't block the PR on it though.